tests/unit/s2n_connection_test.c

```diff
@@ -162,5 +163,82 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(server_conn));
     }
 
+    /* Test: get selected digest alg */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+        s2n_handshake_hash_algorithm output = { 0 };
+
+        EXPECT_FAILURE(s2n_connection_get_selected_digest_algorithm(NULL, &output));
+        EXPECT_FAILURE(s2n_connection_get_selected_digest_algorithm(conn, NULL));
+        EXPECT_FAILURE(s2n_connection_get_selected_client_cert_digest_algorithm(NULL, &output));
+        EXPECT_FAILURE(s2n_connection_get_selected_client_cert_digest_algorithm(conn, NULL));
+
+        EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_digest_algorithm(conn, &output));
+        EXPECT_EQUAL(S2N_HANDSHAKE_HASH_NONE, output);
+
+        EXPECT_SUCCESS(s2n_connection_get_selected_digest_algorithm(conn, &output));
+        EXPECT_EQUAL(S2N_HANDSHAKE_HASH_NONE, output);
+
+        s2n_hash_algorithm inputs[] = { S2N_HASH_MD5, S2N_HASH_SHA1, S2N_HASH_SHA224, S2N_HASH_SHA256, S2N_HASH_SHA384, 
+                                        S2N_HASH_SHA512, S2N_HASH_MD5_SHA1 };
```

Comment:
This works, but is there any way to ensure your test covers every possible hash and signature alg? What if I add a new one?

For hash algorithms, you have S2N_HASH_SENTINEL. You can make sure that you can translate every hash between S2N_HASH_NONE and S2N_HASH_SENTINEL.

For signature algorithms, it's a little trickier. You could try defining a sentinel for s2n_signature_algorithm. Alternatively, you could get a hold of all possible signature algorithms allowed by our preferences, ether by assuming [security_policy_test_all](https://github.com/aws/s2n-tls/blob/fe8df74ddafb712d3141056107db7228b19b6ef7/tls/s2n_security_policies.c#L468-L474) includes them all, or by [iteration over all policies](https://github.com/aws/s2n-tls/blob/fe8df74ddafb712d3141056107db7228b19b6ef7/tls/s2n_security_policies.c#L532), but that seems a little overkill.

Reply:
Added a sentinel value to the signature enum, and added a check for both the hash and signature cases that we are checking all the enum values